### PR TITLE
hook up actionmailer to some environment variables to send email via smtp

### PR DESCRIPTION
**Pull request by pcai**
This PR is part of https://github.com/ReadyResponder/ReadyResponder/issues/199

Adds dotenv gem 
Adds appropriate gitignore changes
Adds SMTP email support, configurable via environment variables

Here's a snippet of my `.env` in my development environment, using a sandbox mailgun account:
```
MAIL_METHOD=smtp
MAIL_SMTP_ADDR=smtp.mailgun.org
MAIL_SMTP_PORT=587
MAIL_SMTP_DOMAIN=sandbox510cbc9b5a074f459e51eb176e7d54d7.mailgun.org
MAIL_USERNAME=dev@sandbox510cbc9b5a074f459e51eb176e7d54d7.mailgun.org
MAIL_PASSWORD={not displayed here}
```

I propose this PR as it is a significant first step towards any email notification system, and I recommend this because:
* ActionMailer can do most of the heavy lifting. In fact, devise hooks into ActionMailer so I was able to send myself a password reset email without problems, in development.
* This setup is completely provider-agnostic - theoretically we could switch between mailgun, sendgrid, SES, or any number of providers by changing config variables.

A couple drawbacks to not using a native gem right away:
* Performance, if the difference between getting an email to inbox in 5s vs 10s is significant. I don't get the sense that it is. Using the SMTP protocol is generally slower performance-wise than the equivalent REST api. 
* `content-transfer-encoding` (etc) and other warts of the SMTP protocol sometimes leak through the abstraction.
* To process inbound emails or events from our provider, we need provider-specific logic anyway, since these tend to be in the form of webhooks (new inbound email, email clicked, email opened, etc).

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -39,12 +39,13 @@
...
+  # Use :letter_opener (open in browser), or :smtp (to test using real SMTP)
+  # Must be a symbol or it wont override action_mailer defaults.
+  config.action_mailer.delivery_method = ENV.fetch("MAIL_METHOD", "letter_opener").to_sym
   config.action_mailer.smtp_settings = {
-    address: "smtp.gmail.com",
-    port: 587,
-    domain: "billericaema.org",
+    address: ENV["MAIL_SMTP_ADDR"],
+    port: ENV["MAIL_SMTP_PORT"],
+    domain: ENV["MAIL_SMTP_DOMAIN"],
```
Comment (houhoulis) [inline config/environments/development.rb:48]: 👍

**Comment by kgf** [comment]
@pcai, thanks for your help.
Is it your expectation that this PR would not have any effect on production sites, as it's part of the solution?
Kevin

**Comment by kgf** [comment]
@pcai This PR looks good, but I was hoping you could answer the question above before merge.

**Comment by pcai** [comment]
Sorry I missed your prior message. My expectation is that it will have no effect on production sites, given the following assumption:

* production sites currently do not send emails of any kind.

I am making this assumption because there are no settings in `production.rb` that I can see that would allow production to send emails

**Comment by kgf** [comment]
Thanks @pcai

**Pull request merged**